fili-core/src/main/java/com/yahoo/bard/webservice/web/endpoints/EndpointServlet.java

```diff
@@ -46,43 +74,124 @@ public EndpointServlet(ObjectMappersSuite objectMappers) {
      */
     protected <T> Response formatResponse(
             ApiRequest apiRequest,
+            Response.ResponseBuilder builder,
+            Pagination pagination,
+            ContainerRequestContext containerRequestContext,
             Stream<T> rows,
             String jsonName,
             List<String> csvColumnNames
     ) {
+        UriInfo uriInfo = containerRequestContext.getUriInfo();
         StreamingOutput output;
-        Response.ResponseBuilder builder;
-        switch (apiRequest.getFormat()) {
-            case CSV:
-                builder = apiRequest.getBuilder()
-                        .header(HttpHeaders.CONTENT_TYPE, "text/csv; charset=utf-8")
-                        .header(
-                                HttpHeaders.CONTENT_DISPOSITION,
-                                ResponseFormat.getCsvContentDispositionValue(apiRequest.getUriInfo())
-                        );
-                output = new CsvResponse<>(
-                        rows,
-                        apiRequest.getPagination(),
-                        apiRequest.getUriInfo(),
-                        csvColumnNames,
-                        objectMappers
-                ).getResponseStream();
-                break;
-            case JSON:
-                // Fall-through: Default is JSON
-            default:
-                builder = apiRequest.getBuilder()
-                        .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON + "; charset=utf-8");
-
-                output = new JsonResponse<>(
-                        rows,
-                        apiRequest.getPagination(),
-                        apiRequest.getUriInfo(),
-                        jsonName,
-                        objectMappers
-                ).getResponseStream();
-                break;
+        if (CSV.accepts(apiRequest.getFormat().toString())) {
```

Comment:
Is the `toString` needed? Can it be resolved to a type in the `apiRequest` so that stringly typed things are only at the boundary?